### DISCUSSION TITLE
assert error when I set my num_mels to 220 with Tacotron2 +ddc

### DISCUSSION THREAD

**Discussion by KakoozaJerry** [coqui-ai/TTS#1281] [General Q&A]
I set my config file as below with num_mels as 220 but I get this error

```
Traceback (most recent call last):
  File "/content/drive/MyDrive/Emergent/train/Day23/train_tacotron_ddc.py", line 77, in <module>
    datasets=[dataset_config],
  File "<string>", line 102, in __init__
  File "/usr/local/lib/python3.7/dist-packages/coqpit/coqpit.py", line 589, in __post_init__
    self.check_values()
  File "/content/TTS/TTS/tts/configs/tacotron_config.py", line 225, in check_values
    assert self.out_channels == self.audio.num_mels
AssertionError
```

Config setup
```
audio_config = BaseAudioConfig(
    sample_rate=22050,
    do_trim_silence=False,
    trim_db=60.0,
    power=1.0,
    signal_norm=False,
    num_mels=220,
    mel_fmin=0.0,
    mel_fmax=8000,
    spec_gain=1.0,
    log_func="np.log",
    ref_level_db=20,
    preemphasis=0.98,
    do_sound_norm=True,
)

config = Tacotron2Config(  # This is the config that is saved for the future use
    audio=audio_config,
    run_name="Tacotron2_DDC_220mels_1power_0_98_preemph_soundNorm",
    batch_size=32,
    eval_batch_size=16,
    num_loader_workers=4,
    num_eval_loader_workers=4,
    run_eval=True,
    test_delay_epochs=-1,
    r=6,
    gradual_training=[[0, 6, 32], [10000, 4, 32], [50000, 3, 32], [100000, 2, 32], [200000, 1, 16]], 
    double_decoder_consistency=True,
    epochs=10000,
    text_cleaner="basic_cleaners",
    use_phonemes=False,
    phoneme_language="en-us",
    phoneme_cache_path="/content/drive/MyDrive/Emergent/train/Day11_newData/phoneme_cache",
    print_step=25,
    print_eval=True,
    mixed_precision=False,
    save_step=1000,
    characters={
        "pad": "_",
        "eos": "~",
        "bos": "^",
        "characters": "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyzŋ!'(),-.:;? ",
        "punctuations": "!'(),-.:;? ",
        "phonemes": "iy\u0268\u0289\u026fu\u026a\u028f\u028ae\u00f8\u0258\u0259\u0275\u0264o\u025b\u0153\u025c\u025e\u028c\u0254\u00e6\u0250a\u0276\u0251\u0252\u1d7b\u0298\u0253\u01c0\u0257\u01c3\u0284\u01c2\u0260\u01c1\u029bpbtd\u0288\u0256c\u025fk\u0261q\u0262\u0294\u0274\u014b\u0272\u0273n\u0271m\u0299r\u0280\u2c71\u027e\u027d\u0278\u03b2fv\u03b8\u00f0sz\u0283\u0292\u0282\u0290\u00e7\u029dx\u0263\u03c7\u0281\u0127\u0295h\u0266\u026c\u026e\u028b\u0279\u027bj\u0270l\u026d\u028e\u029f\u02c8\u02cc\u02d0\u02d1\u028dw\u0265\u029c\u02a2\u02a1\u0255\u0291\u027a\u0267\u02b2\u025a\u02de\u026b",
        "unique": True
    },
    test_sentences=[
        "Abaaliwo bonna baasooka ne basiriikirira nga beewuunya omulanga oguvudde mu musajja ayogereza mwannyinaabwe.",
        "Wuuno eyakubye bbomu mu bbaasi.",
        "Engeri omupangisa gy’atutte omugagga nabukeera mu kkomera.",
        "Pulezidenti awadde ekiragiro ku takisi.",
        "Aba bodaboda abasse ow’emmotoka babalina."
    ],
    max_decoder_steps=2000,
    min_seq_len=50,
    max_seq_len=500000,
    output_path=output_path,
    datasets=[dataset_config],
)
```

**Comment by catalinnega** [comment]
You should set the "out_channels" parameter in Tacotron2Config to match the number of mels.

**Reply by KakoozaJerry** [reply]
I set it and get another error

```
 > Setting up Audio Processor...
 | > sample_rate:22050
 | > resample:False
 | > num_mels:220
 | > log_func:np.log
 | > min_level_db:-100
 | > frame_shift_ms:None
 | > frame_length_ms:None
 | > ref_level_db:20
 | > fft_size:1024
 | > power:1.0
 | > preemphasis:0.98
 | > griffin_lim_iters:60
 | > signal_norm:False
 | > symmetric_norm:True
 | > mel_fmin:0
 | > mel_fmax:8000
 | > spec_gain:1.0
 | > stft_pad_mode:reflect
 | > max_norm:4.0
 | > clip_norm:True
 | > do_trim_silence:False
 | > trim_db:60.0
 | > do_sound_norm:True
 | > do_amp_to_db_linear:True
 | > do_amp_to_db_mel:True
 | > do_rms_norm:False
 | > db_level:None
 | > stats_path:None
 | > base:2.718281828459045
 | > hop_length:256
 | > win_length:1024
 | > Found 17131 files in /content/data
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
 > Using CUDA:  True
 > Number of GPUs:  1

 > Model has 48353944 parameters

 > Number of output frames: 6

 > EPOCH: 0/10000
 --> /content/drive/MyDrive/Emergent/train/Day23/Tacotron2_DDC_220mels_1power_0_98_preemph_soundNorm-February-23-2022_11+28AM-0000000

 > DataLoader initialization
 | > Use phonemes: False
 | > Number of instances : 16960
 | > Max length sequence: 148
 | > Min length sequence: 8
 | > Avg length sequence: 58.44009433962264
 | > Num. instances discarded by max-min (max=500000, min=50) seq limits: 6359
 | > Batch group size: 0.
/usr/local/lib/python3.7/dist-packages/torch/utils/data/dataloader.py:481: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  cpuset_checked))

 > TRAINING (2022-02-23 11:28:43) 
Traceback (most recent call last):
  File "/content/TTS/TTS/trainer.py", line 1007, in fit
    self._fit()
  File "/content/TTS/TTS/trainer.py", line 992, in _fit
    self.train_epoch()
  File "/content/TTS/TTS/trainer.py", line 820, in train_epoch
    _, _ = self.train_step(batch, batch_num_steps, cur_step, loader_start_time)
  File "/content/TTS/TTS/trainer.py", line 691, in train_step
    batch, self.model, self.optimizer, self.scaler, self.criterion, self.scheduler, self.config
  File "/content/TTS/TTS/trainer.py", line 601, in _optimize
    outputs, loss_dict = self._model_train_step(batch, model, criterion)
  File "/content/TTS/TTS/trainer.py", line 560, in _model_train_step
    return model.train_step(*input_args)
  File "/content/TTS/TTS/tts/models/tacotron2.py", line 263, in train_step
    aux_input={"speaker_ids": speaker_ids, "d_vectors": d_vectors},
  File "/content/TTS/TTS/tts/models/tacotron2.py", line 168, in forward
    decoder_outputs, alignments, stop_tokens = self.decoder(encoder_outputs, mel_specs, input_mask)
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/content/TTS/TTS/tts/layers/tacotron/tacotron2.py", line 312, in forward
    memories = torch.cat((memory, memories), dim=0)
RuntimeError: Sizes of tensors must match except in dimension 0. Expected size 480 but got size 220 for tensor number 1 in the list.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/content/drive/MyDrive/Emergent/train/Day23/train_tacotron_ddc.py", line 100, in <module>
    trainer.fit()
  File "/content/TTS/TTS/trainer.py", line 1026, in fit
    remove_experiment_folder(self.output_path)
  File "/content/TTS/TTS/utils/generic_utils.py", line 74, in remove_experiment_folder
    print(" ! Run is removed from {}".format(experiment_path))
  File "/content/TTS/TTS/trainer.py", line 1184, in write
    with open(self.log_file, "a", encoding="utf-8") as f:
FileNotFoundError: [Errno 2] No such file or directory: '/content/drive/MyDrive/Emergent/train/Day23/Tacotron2_DDC_220mels_1power_0_98_preemph_soundNorm-February-23-2022_11+28AM-0000000/trainer_0_log.txt'
```

**Reply by vamsi-eshipjet** [reply]
> I set it and get another error
> 
> ```
>  > Setting up Audio Processor...
>  | > sample_rate:22050
>  | > resample:False
>  | > num_mels:220
>  | > log_func:np.log
>  | > min_level_db:-100
>  | > frame_shift_ms:None
>  | > frame_length_ms:None
>  | > ref_level_db:20
>  | > fft_size:1024
>  | > power:1.0
>  | > preemphasis:0.98
>  | > griffin_lim_iters:60
>  | > signal_norm:False
>  | > symmetric_norm:True
>  | > mel_fmin:0
>  | > mel_fmax:8000
>  | > spec_gain:1.0
>  | > stft_pad_mode:reflect
>  | > max_norm:4.0
>  | > clip_norm:True
>  | > do_trim_silence:False
>  | > trim_db:60.0
>  | > do_sound_norm:True
>  | > do_amp_to_db_linear:True
>  | > do_amp_to_db_mel:True
>  | > do_rms_norm:False
>  | > db_level:None
>  | > stats_path:None
>  | > base:2.718281828459045
>  | > hop_length:256
>  | > win_length:1024
>  | > Found 17131 files in /content/data
> fatal: not a git repository (or any of the parent directories): .git
> fatal: not a git repository (or any of the parent directories): .git
>  > Using CUDA:  True
>  > Number of GPUs:  1
> 
>  > Model has 48353944 parameters
> 
>  > Number of output frames: 6
> 
>  > EPOCH: 0/10000
>  --> /content/drive/MyDrive/Emergent/train/Day23/Tacotron2_DDC_220mels_1power_0_98_preemph_soundNorm-February-23-2022_11+28AM-0000000
> 
>  > DataLoader initialization
>  | > Use phonemes: False
>  | > Number of instances : 16960
>  | > Max length sequence: 148
>  | > Min length sequence: 8
>  | > Avg length sequence: 58.44009433962264
>  | > Num. instances discarded by max-min (max=500000, min=50) seq limits: 6359
>  | > Batch group size: 0.
> /usr/local/lib/python3.7/dist-packages/torch/utils/data/dataloader.py:481: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
>   cpuset_checked))
> 
>  > TRAINING (2022-02-23 11:28:43) 
> Traceback (most recent call last):
>   File "/content/TTS/TTS/trainer.py", line 1007, in fit
>     self._fit()
>   File "/content/TTS/TTS/trainer.py", line 992, in _fit
>     self.train_epoch()
>   File "/content/TTS/TTS/trainer.py", line 820, in train_epoch
>     _, _ = self.train_step(batch, batch_num_steps, cur_step, loader_start_time)
>   File "/content/TTS/TTS/trainer.py", line 691, in train_step
>     batch, self.model, self.optimizer, self.scaler, self.criterion, self.scheduler, self.config
>   File "/content/TTS/TTS/trainer.py", line 601, in _optimize
>     outputs, loss_dict = self._model_train_step(batch, model, criterion)
>   File "/content/TTS/TTS/trainer.py", line 560, in _model_train_step
>     return model.train_step(*input_args)
>   File "/content/TTS/TTS/tts/models/tacotron2.py", line 263, in train_step
>     aux_input={"speaker_ids": speaker_ids, "d_vectors": d_vectors},
>   File "/content/TTS/TTS/tts/models/tacotron2.py", line 168, in forward
>     decoder_outputs, alignments, stop_tokens = self.decoder(encoder_outputs, mel_specs, input_mask)
>   File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/module.py", line 1102, in _call_impl
>     return forward_call(*input, **kwargs)
>   File "/content/TTS/TTS/tts/layers/tacotron/tacotron2.py", line 312, in forward
>     memories = torch.cat((memory, memories), dim=0)
> RuntimeError: Sizes of tensors must match except in dimension 0. Expected size 480 but got size 220 for tensor number 1 in the list.
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/content/drive/MyDrive/Emergent/train/Day23/train_tacotron_ddc.py", line 100, in <module>
>     trainer.fit()
>   File "/content/TTS/TTS/trainer.py", line 1026, in fit
>     remove_experiment_folder(self.output_path)
>   File "/content/TTS/TTS/utils/generic_utils.py", line 74, in remove_experiment_folder
>     print(" ! Run is removed from {}".format(experiment_path))
>   File "/content/TTS/TTS/trainer.py", line 1184, in write
>     with open(self.log_file, "a", encoding="utf-8") as f:
> FileNotFoundError: [Errno 2] No such file or directory: '/content/drive/MyDrive/Emergent/train/Day23/Tacotron2_DDC_220mels_1power_0_98_preemph_soundNorm-February-23-2022_11+28AM-0000000/trainer_0_log.txt'
> ```

Hey. Were you able to solve this?
